### PR TITLE
misc tosca fixes

### PR DESCRIPTION
**Pull request by ahgittin**
* better error reporting
* support for TOSCA plans in entity spec (requires update to Brooklyn which is not yet merged)

### CODE REVIEW

File: brooklyn-tosca-transformer/src/test/java/io/cloudsoft/tosca/a4c/brooklyn/plan/ToscaTypePlanTransformerIntegrationTest.java [hunk 1]
```diff
@@ -70,6 +75,8 @@
 
 public class ToscaTypePlanTransformerIntegrationTest extends Alien4CloudIntegrationTest {
```
Comment (iuliana) [inline brooklyn-tosca-transformer/src/test/java/io/cloudsoft/tosca/a4c/brooklyn/plan/ToscaTypePlanTransformerIntegrationTest.java:76]: This class is getting  quite big. Maybe we should consider splitting it based on scope (negative/positive tests - the samples could be grouped too) or by any other category.

File: brooklyn-tosca-transformer/src/test/resources/templates/simple-with-yaml-error.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
+
```
Comment (iuliana) [inline brooklyn-tosca-transformer/src/test/resources/templates/simple-with-yaml-error.yaml:1]: This file contains a Yaml indentation error and the output in the console is identical to the previous one that has a relationship referencing a missing node.  Isn't there a way to print an error message that is obvious  about its cause?

File: brooklyn-tosca-transformer/src/test/java/io/cloudsoft/tosca/a4c/brooklyn/plan/ToscaTypePlanTransformerIntegrationTest.java (hunk 1 above)
Comment (ahgittin) [reply brooklyn-tosca-transformer/src/test/java/io/cloudsoft/tosca/a4c/brooklyn/plan/ToscaTypePlanTransformerIntegrationTest.java:76]: agree -- but not here

File: brooklyn-tosca-transformer/src/test/resources/templates/simple-with-yaml-error.yaml (hunk 2 above)
Comment (ahgittin) [reply brooklyn-tosca-transformer/src/test/resources/templates/simple-with-yaml-error.yaml:1]: the error it prints is:

```
org.apache.brooklyn.util.exceptions.PropagatedRuntimeException: Unable to instantiate item: Plan looks like it's meant to be TOSCA but it is not valid YAML: ScannerException: mapping values are not allowed here
 in 'reader', line 16, column 17:
      node_templates:
```

**Pull request merged**